packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -340,7 +340,10 @@ export class EcsRunTask extends sfn.TaskStateBase implements ec2.IConnectable {
     const policyStatements = [
       new iam.PolicyStatement({
         actions: ['ecs:RunTask'],
-        resources: [this.getTaskDefinitionFamilyArn()],
+        resources: [
+          this.getTaskDefinitionFamilyArn(),
+          `${this.getTaskDefinitionFamilyArn()}:*`,
```

Comment:
Drawing attention to the bits that add permission to run versions of the task definition.